ui/sign_in_spec.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+describe 'Sign in page' do
+  context 'when unauthenticated' do
+    include_context 'as guest'
+
+    before do
+      ui.sign_in.load
+    end
+
+    it 'Warning is displayed when the user clicks login with blank username' do
+      ui.sign_in.with_email_and_password('', '')
+      expect(ui.sign_in).to have_warning_title
+    end
+
+    it 'Warning is displayed when the user clicks login with invalid username' do
+      ui.sign_in.with_email_and_password(nil, 'Strongassword')
+      expect(ui.sign_in).to have_warning_title
+    end
+
+    it 'is shown expanded Help menu' do
+      sleep(10)
+      ui.sign_in.problem_signing_in.click
+      expect(ui.sign_in).to have_content 'I have forgotten my password'
+      expect(ui.sign_in).to have_content 'receive confirmation instructions?'
+      expect(ui.sign_in).to have_content 'Other problems signing in'
+    end
+
+    it 'is shown I have forgotten my Password page' do
```

Comment:
We can then do away with this block because we have asserted the link, we needn't bother visiting it. The `forgotten_my_password` spec can then be responsible for ensuring the header is present.

Reply:
Or is it preferable to couple pages this way and not have a spec for that page by itself?